tools/snapshot/internal/export/export.go

```diff
@@ -0,0 +1,245 @@
+	var byteArray [32]byte
+
+	incr := 256 / markerNum
+
+	var byteArrayList [][32]byte
+
+	for i := 0; i < int(markerNum); i++ {
+		byteArray[0] = byte(i * int(incr)) // Increment the highest byte
+		byteArrayList = append(byteArrayList, byteArray)
+	}
+
+	return byteArrayList
+}
+
+func saveLedgerData(client pb.XRPLedgerAPIServiceClient, seq uint32, marker []byte, end []byte, path string) {
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	request := pb.GetLedgerDataRequest{}
+	ledger := &pb.LedgerSpecifier{
+		Ledger: &pb.LedgerSpecifier_Sequence{
+			Sequence: seq,
+		},
+	}
+	request.Ledger = ledger
+	request.Marker = marker[:]
+	if end != nil {
+		request.EndMarker = end[:]
+	}
+	request.User = grpcUser
+
+	subPath := filepath.Join(path, fmt.Sprintf("ledger_data_%d", seq), fmt.Sprintf("marker_%x", marker))
+	err := os.MkdirAll(subPath, os.ModePerm)
+	if err != nil {
+		log.Fatalf("Error creating directory: %v", err)
+	}
+
+	for request.Marker != nil {
+		res, err := client.GetLedgerData(ctx, &request)
+		if err != nil {
+			log.Fatalf("Error getting ledger data: %v", err)
+		}
+
+		filePath := filepath.Join(subPath, fmt.Sprintf("%x.dat", request.Marker))
+
+		protoData, err := proto.Marshal(res)
+		if err != nil {
+			log.Fatalf("Error marshalling data: %v", err)
+		}
+
+		err = os.WriteFile(filePath, protoData, 0644)
+		if err != nil {
+			log.Fatalf("failed to write file: %v", err)
+		}
+
+		request.Marker = res.Marker
+	}
+
+}
+
+func getLedgerFullData(client pb.XRPLedgerAPIServiceClient, seq uint32, path string) {
+	log.Printf("Processing full sequence: %d\n", seq)
+
+	markers := generate256Markers(markerNum)
+
+	var wg sync.WaitGroup
+
+	// Launch a goroutine for each marker
+	for i, marker := range markers {
+		wg.Add(1)
+		var end []byte
+		if i != len(markers)-1 {
+			end = markers[i+1][:]
+		}
+
+		fmt.Printf("Got ledger data marker: %x-%x\n", marker, end)
+
+		go func(marker []byte, end []byte) {
+			defer wg.Done()
+			saveLedgerData(client, seq, marker, end, path)
+		}(marker[:], end)
+
+	}
+
+	wg.Wait()
+
+}
+
+func checkPath(path string) {
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		// Create the directory if it doesn't exist
+		err := os.MkdirAll(path, os.ModePerm)
+		if err != nil {
+			log.Fatalf("Error creating directory: %v", err)
+		}
+	}
+}
+
+func ExportFromFullLedger(grpcServer string, startSeq uint32, endSeq uint32, path string) {
+	checkPath(path)
+
+	client, err := createGRPCClient(grpcServer)
+	if err != nil {
+		log.Fatalf("Error creating gRPC client: %v", err)
+	}
+
+	defer client.Close()
+
+	ExportFromFullLedgerImpl(client.Client, startSeq, endSeq, path)
+
+}
+
+func ExportFromFullLedgerImpl(client pb.XRPLedgerAPIServiceClient, startSeq uint32, endSeq uint32, path string) {
+
+	getLedgerFullData(client, startSeq, path)
+
+	//We need to fetch the ledger header and txs for startSeq as well
+	for i := startSeq; i <= endSeq; i++ {
```

Comment:
Shouldn't it be `startSeq + 1` because we already go full data for `startSeq`?

Reply:
Ledger full data does not contain the header and transactions.